src/haz3lweb/view/ScratchMode.re

```diff
@@ -95,6 +96,18 @@ module Update = {
     | CellAction(a) =>
       let (key, ed) = List.nth(model.scratchpads, model.current);
       let* new_ed = CellEditor.Update.update(~settings, a, ed);
+      switch (a) {
+      | MainEditor(action) =>
+        switch (action) {
+        | Perform(a) =>
```

Comment:
just match directly on `| Perform(Insert(char))`. maybe make this a helper with a reasonable name, or else add a comment